### PR TITLE
fix(infrastructure): Removing PII info from URLs when sending to DataDog

### LINKED ISSUE #4969: We need to avoid logging the kennitala in DD traces when invoking the RLS driver's license API.
This is critical for elections.

We still need to see RLS api calls in DD for alerting and what not.

### PR DESCRIPTION
**Pull request by pshomov**
# ...

Fixes #4969 and fixes some feature-branch deployment problems while at it.

## What

See issue.

## Why

See issue.

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Formatting passes locally with my changes
- [ ] I have rebased against main before asking for a review

### CODE REVIEW

File: libs/infra-tracing/src/lib/datadog-tracer.ts [hunk 1]
```diff
@@ -1,8 +1,27 @@
...
+  tracer.use('http', {
+    client: {
+      hooks: {
+        request: (span, req, res) => {
+          const url = res?.url
+          if (url) {
+            const urlRewritten = url
+              .split('/')
+              .map((segm) =>
+                isPerson(segm) && segm.length === 10 ? '--MASKED--' : segm,
```
Comment (sindrig) [inline libs/infra-tracing/src/lib/datadog-tracer.ts:16]: length check is performed by `kennitala` and is not needed here

**Review by sindrig — COMMENTED** [review]
Testing out in feature deployment

File: libs/infra-tracing/src/lib/datadog-tracer.ts (hunk 1 above)
Comment (pshomov) [reply libs/infra-tracing/src/lib/datadog-tracer.ts]: 👍🏻 fixed

File: libs/infra-tracing/src/lib/datadog-tracer.ts [hunk 2]
```diff
@@ -1,8 +1,26 @@
...
 if (process.env.NODE_ENV !== 'development') {
   tracer.init({ logInjection: true }) // initialized in a different file to avoid hoisting.
   tracer.use('express', { blacklist: ['/liveness', '/readiness', '/metrics'] })
+  tracer.use('http', {
+    client: {
+      hooks: {
+        request: (span, req, res) => {
+          const url =
+            res?.url || `http(s)://${req?.getHeader('host')}${req?.path}`
+          if (url) {
```
Comment (sindrig) [inline libs/infra-tracing/src/lib/datadog-tracer.ts:13]: this check is probably not needed anymore since url can't be falsy.

**Review by sindrig — CHANGES_REQUESTED** [review]

File: libs/infra-tracing/src/lib/datadog-tracer.ts (hunk 2 above)
Comment (pshomov) [reply libs/infra-tracing/src/lib/datadog-tracer.ts]: Good point, fixed

**Review by sindrig — APPROVED** [review]

**Pull request merged**